### ISSUE TITLE
Issue with Kubernetes DNS resolution after deploying GlusterFS

### ISSUE THREAD

**Issue by kevinpz** [gluster/gluster-kubernetes#424]
Hi guys,

I'm facing a little issue, using the `gk-deploy` tool to deploy a glusterFS cluster into an existing K8S cluster. (deployed with kubeadm)

My cluster is working, as well as the K8S dns. If I do a `nslookup google.fr` from a centos pod, it returns me the right IP.
But as soon as I start the `gk-deploy` tool, I'm facing issue with the DNS resolution. If I retry the same command, it answers me with the following error: `reply from unexpected source`, showing me the IP of the DNS POD instead of the DNS service IP. (which seems to be a masquerade error)

I tried to investigate a little bit, and the issue disappears when I restart on all my workers the `weaveworks/weave-kube` container.

But I didn't find what is the root cause of this issue, and I'm pretty sure it is caused by the glusterFS deployment, because it doesn't appear until I run the script.

The deployment is pretty simple, one K8S master, 3 K8S workers, each one with a 100GB disk.

I can provide any kind of logs if it can help, thanks !

**Comment by jarrpa** [comment]
...wow, I'm not sure where to start. This is bizarre! The only thing I can imagine that might be related is the fact that the GlusterFS pods use hostNetwork networking. Otherwise they don't do anything to mess with any networking configuration. Can you try to replicate this by spinning up a simple nginx pod on each node that uses `hostNetwork: true`?

**Comment by zrahui** [comment]
I've also observed the same behaviour in a k8s cluster built with ACS Engine using Azure CNI. Deploying with hostNetwork seems to kill the DNS in the cluster. Removing hostNetwork from the yaml works fine, but then raises another issue of being able to know the IP assigned to gluster pods during deployment.

**Comment by jarrpa** [comment]
@zrahui Hmm... can you think of any workaround? This sounds significant enough that we should probably extend the help prompt at the start of the script.

**Comment by zrahui** [comment]
@jarrpa Unfortunately i'm stuck on issue #432 so haven't been able to get much further. If I get things working I'll let you know!

**Comment by zrahui** [comment]
It's definitely the gk-deploy tool, specifically the spinning up of the GlusterFS pods that seems to kill DNS. I've managed to run a DS with busybox and hostNetwork enabled with no issues. I'm at a loss as to where the issue lies, basically the internal kube-dns falls over and won't come back. 
The kube-dns logs are specifying an i/o timeout to apiserver. It's like the routes to the apiserver have been lost. 
Any ideas on where to check?

**Comment by jarrpa** [comment]
There have been some issues on clusters deployed with... I think kubespray? The issue was that kubespray had some sort of port range that overlapped with some ports used by GlusterFS. I'll try to find the issue.

**Comment by jarrpa** [comment]
Yes, kubespray: https://github.com/gluster/gluster-kubernetes/issues/286#issuecomment-329154392

Please open a new Issue if problems persist.

**Comment by zrahui** [comment]
Removing the gluster DS, clearing the /var/lib/glusterd directory and rebooting each node brings connectivity and DNS back online. So something in gluster is conflicting with the ACS configuration. I'm at a loss as to where to look next so have logged an issue with ACS-engine team https://github.com/Azure/acs-engine/issues/2222

**Comment by kevinpz** [comment]
Hi @jarrpa , sure I'm still interested in this issue. It's really annoying to have to restart Docker in all of my workers after installing gluster.

For information, I'm not using kubespray, but kubeadm. And I'm deploying on an OpenStack cloud private platform.

**Comment by jarrpa** [comment]
@kevinpz All right, good to know. You mentioned earlier that you restarted the weave-kube pods, not Docker itself. Do both of these seem to have the same effect? Can you check to see what the value of `net.ipv4.ip_local_reserved_ports` on your hosts is when you're having the networking problem (i.e. before restarting anything)?
